yamui-screensaverd.c

```diff
@@ -0,0 +1,361 @@
+/*
+ * Simple screen saver daemon. Turns off the display after idle timeout.
+ * Turns the display on on any event from /dev/input/event* files.
+ * On exit turns the display on.
+ *
+ * Copyright (C) 2015 Jolla Ltd.
+ * Contact: Igor Zhbanov <igor.zhbanov@jolla.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define _BSD_SOURCE
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <dirent.h>
+#include <signal.h>
+#include <stdarg.h>
+#include <string.h>
+#include <unistd.h>
+#include <linux/input.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <sys/select.h>
+
+#define UNUSED			__attribute__((unused))
+#define NBITS(x)		((((x) - 1) / __BITS_PER_LONG) + 1)
+#define BIT(arr, bit)		((arr[(bit) / __BITS_PER_LONG] >> \
+				 ((bit) % __BITS_PER_LONG)) & 1)
+
+/*#define DEBUG*/
+
+#define DISPLAY_CONTROL		"/sys/class/graphics/fb0/blank"
+#define DEV_INPUT_DIR		"/dev/input"
+#define LOG_NAME		"screensaverd"
+#define EVENT_PREFIX		"event"
+#define MAX_DEVICES		256
+#define DISPLAY_OFF_TIME	25 /* seconds */
+#define EVENTS_BUF_SIZE		512 /* events */
+#define EXIT_SUCCESS		0
+#define EXIT_FAILURE		1
+
+sig_atomic_t volatile running = 1;
+
+/* ------------------------------------------------------------------------ */
+
+static void
+print_common(const char *format, va_list ap)
+{
+	printf("[%s] ", LOG_NAME);
+	vprintf(format, ap);
+}
+
+/* ------------------------------------------------------------------------ */
+
+static void
+infof(const char *format, ...) __attribute__ ((format (printf, 1, 2)));
+
+/* Info printing with printf-style format and application name prefix. */
+static void
+infof(const char *format, ...)
+{
+	va_list ap;
+	int old_errno = errno;
+
+	va_start(ap, format);
+	print_common(format, ap);
+	printf("\n");
+	va_end(ap);
+	errno = old_errno; /* Just in case. */
+}
+
+/* ------------------------------------------------------------------------ */
+
+#ifdef DEBUG
+
+#define debugf	infof
+
+#else /* !DEBUG */
+
+static void
+debugf(const char *format, ...) __attribute__ ((format (printf, 1, 2)))
+				UNUSED;
+
+static void
+debugf(const char *format UNUSED, ...)
+{
```

Comment:
In Linux kernel they prefer to have stubs doing nothing vs eliminating it by the means of preprocessor. Always validating arguments could be useful because a programmer can commit wrong debug line didn't noticed it because of disabled debug. And some other programmer could realize that the application doesn't work or even compile with debug is on.